www/js/app.js

```diff
@@ -358,7 +384,13 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
     document.getElementById('appThemeSelect').addEventListener('change', function (e) {
         params.appTheme = e.target.value;
         settingsStore.setItem('appTheme', params.appTheme, Infinity);
-        uiUtil.applyAppTheme(params.appTheme);
+        if (params.appTheme === 'auto_invert') {
+            handleAutoGenTheme();
+        }
+        else if (params.appTheme === 'auto_mwInvert') {
+            handleAutoWikiTheme();
+        }
+        else uiUtil.applyAppTheme(params.appTheme);
```

Comment:
Ditto. None of this is necessary. You need to handle things generically in the one function in applyAppTheme.

Reply:
Done